docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,176 @@
+---
+sidebar_label: glides.vehicle_trip_assignments
+sidebar_position: 2
+---
+
+import EventSchemaPath from "@site/src/components/EventSchemaPath.tsx";
+import Examples from "@site/src/components/Examples.tsx";
+import example from "@site/examples/glides-events/vehicle_trip_assignments.v1.json";
+
+# vehicle_trip_assignments
+
+event type: `com.mbta.ctd.glides.vehicle_trip_assignments.v1`
+
+Describes which vehicle is operating on which trip at the current moment. An event is sent whenever a vehicle is assigned or unassigned from a trip. One event may batch multiple assignments.
+
+Does not provide predictions for which vehicles will be assigned to future trips.
+
+See the the [Updating Assignments](#updating-assignments) section for how vehicles and trips are assigned and unassigned.
+
+## Data
+
+Fields in the event's `data`:
+
+- `assignments` (array of [VehicleTripAssignment](#vehicletripassignment)): The list of vehicles and trips with new assignments.
```

Comment:
**question:** how great do we feel the benefit is of including an array of `assignments` as opposed to each event containing a single assignment update? It would save us some bytes when Glides updates multiple vehicles - trip assignments at once, which I guess could happen when new realtime data comes in, but assignment updates from trainsheet edits would probably tend to come one at a time. I think especially if we've reworked the semantics such that the state will be consistent after processing each individual element of the assignments array, we should at least consider just making the structure flat.

Reply:
For my 2c, I'd prefer a single event per. That frees us up on the receiving end to process them one by one vs a batch of n, and I think makes more sense thinking of this from an event-driven perspective.

Reply:
👍 We'll make them as a batch when we get new vehicle positions from GTFS-RT, but also one by one at other times. And the issue about requiring batches in order to be consistent is fixed. If it'll be easier for you, then I'll go ahead and make this change.

Reply:
Done in [0079c2b](https://github.com/mbta/schemas/pull/9/commits/0079c2bdbf488b46dd4e5f1db0f94d74e730a971)